### ISSUE TITLE
Unable to import 'load_openai'PylintE0401:import-error

### ISSUE THREAD

**Issue by RicardoEscobar** [microsoft/vscode-pylint#353]
Pylint is showing false positives on imports. I dunno why.
I'm running a venv and I installed it: pip install pylint 

<h3> Environment data </h3>

-   Language Server version: 2023.6.30
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```python
XXX
```

<h3> Repro Steps </h3>

1. XXX

<h3> Expected behavior </h3>

XXX

<h3> Actual behavior </h3>

XXX

<h3> Logs </h3>

```
XXX
```

**Comment by RicardoEscobar** [comment]
> Hi @RicardoEscobar! Apologies for the delay in our response. Are you using the[ Pylint extension](https://marketplace.visualstudio.com/items?itemName=ms-python.pylint)?
> 
> I'd recommend using that since it ships with pylint, so you don't need to install it in your environment. In any case, I'd recommend running the "Python: Select Interpreter" command from the Command Palette (View > Command Palette...) and selecting the virtual environment you are using in the terminal 😊
> 
> In the mean time I'm going to transfer this issue to the Pylint extension repository

Thank you for answering, I figured that my issue was (and still is) how Python handles imports, specially using PYTHONPATH environment variable. I got rid of the PYTHONPATH environment variable and I'm refactoring to using absolute import paths.

I'm using the `Pylint` extension and ran the  the "Python: Select Interpreter" as you recommended. See below.
![Screenshot_82](https://github.com/microsoft/vscode-pylint/assets/9524910/1cb29af6-3f0e-4aaf-844e-299aa9a79e21)

I'm running a Python virtual environment using the `venv` module.

When using PYTHONPATH environment variable, the app ran successfully but Pylint doesn't know the imports work properly.

Thanks for your time.

**Comment by karthiknadig** [comment]
@RicardoEscobar How are you settings the PYTHONPATH environment variable? using a `.env` file?

It should work if you set your environment variables from terminal.
```
> cd <your project>
> export PYTHONPATH=<paths>
> code . 
```

**Comment by RicardoEscobar** [comment]
the PYTHONPATH environment variable is empty:
![Screenshot_83](https://github.com/microsoft/vscode-pylint/assets/9524910/e24ee340-8b71-4add-9d19-698140abc01e)

I removed it and I'm trying to comply with how Python resolves imports according to [this article: How Imports Work](https://realpython.com/absolute-vs-relative-python-imports/#how-imports-work)

So I'm not setting PYTHONPATH or plan on using it again, it gives more problems than it solves.

**Comment by karthiknadig** [comment]
If you need to set any special locations, you can use `python.analysis.extraPaths`, this will let both `pylance` and `pylint` know where various imports are.
